docs/changelog/3_x.rst

```diff
@@ -0,0 +1,517 @@
+
+    Use this command for installation under Debian/Ubuntu:
+
+    .. code-block:: bash
+
+        sudo apt-get update && sudo apt-get install edgedb-3
+
+    Under CentOS/RHEL, use this installation command:
+
+    .. code-block:: bash
+
+        sudo yum install edgedb-3
+
+    In any required ``systemctl`` commands, replace ``edgedb-server-2`` with
+    ``edgedb-server-3``.
+
+    Under any Docker setups, supply the ``3.0-beta.1`` tag.
+
+2. Take your application offline, then dump your v2.x database with the CLI
+
+   .. code-block:: bash
+
+     $ edgedb dump --dsn <old dsn> --all my_database.dump/
+
+   This will dump the schema and contents of your current database to a file
+   on your local disk called ``my_database.dump``. The file name isn't
+   important.
+
+3. Restore the empty v3.x instance from the dump
+
+   .. code-block:: bash
+
+     $ edgedb restore --all my_database.dump/ --dsn <new dsn>
+
+   Once the restore is complete, update your application to connect to the new
+   instance.
+
+This process will involve some downtime, specifically during steps 2 and 3.
+
+
+Client libraries
+----------------
+
+We've released new versions of our JavaScript and Python client libraries that
+support all 3.0 features and implement the updated protocol. These versions
+are backwards compatible with v1.x and v2.x instances, so we encourage all users to
+upgrade.
+
+.. list-table::
+
+  * - :ref:`TypeScript/JS <edgedb-js-intro>`
+    - ``edgedb@0.21.0``
+  * - :ref:`Python <edgedb-python-intro>`
+    - ``edgedb@0.24.0``
+  * - `Golang <https://www.edgedb.com/docs/clients/go/index>`_
+    - ``edgedb@0.12.0``
+  * - `Rust <https://github.com/edgedb/edgedb-rust>`_
+    - ``edgedb-tokio@0.3.0``
+  * - `.NET <https://github.com/quinchs/EdgeDB.Net>`_ (community-maintained)
+    - ``EdgeDB.Net.Driver@0.3.0``
+  * - `Elixir <https://github.com/nsidnev/edgedb-elixir>`_
+      (community-maintained)
+    - ``edgedb@0.4.0``
+
+New features
+============
+
+Simplified SDL syntax
+---------------------
+
+As part of our commitment to delivering the best developer experience in
+databases, we've made our schema definition language (or SDL) easier to use.
+You're no longer required to use the ``property`` or ``link`` keywords for
+non-computed properties and links. Also, we've replaced arrows with colons for
+a cleaner look that's easier to type.
+
+That means that this type definition:
+
+.. code-block:: sdl
+
+    type User {
+      required property email -> str;
+      multi link friends -> User;
+    }
+
+could be replaced with this equivalent one in EdgeDB 3+:
+
+.. code-block:: sdl
+
+    type User {
+      required email: str;
+      multi friends: User;
+    }
+
+As you're reading through our docs, you'll notice that changing the version
+selector in the sidebar will also switch all SDL examples to the appropriate
+SDL syntax.
+
+.. note::
+
+    SDL syntax in documentation does not change for non-versioned content like
+    guides.
```

Comment:
I'm reading this as "SDL syntax in documentation for non-versioned content like guides will take longer to upgrade to the new syntax", is that right? i.e. probably updated on a best-effort basis but won't be 3.0 syntax starting today.

If that's so I might reword to:

"SDL in documentation like guides is non-versioned content and will be updated to the new syntax over time."

Reply:
That's sorta what I'm trying to say here. I really just want people to know that changing the version dropdown won't affect SDL code in non-versioned content.

Maybe I can write around this by changing the preceding paragraph. Would do you think about this rewrite of that paragraph, @Dhghomon?

> Selecting "v3" from the version dropdown in the sidebar will update any SDL code in versioned sections of the documentation to the new syntax.

Some of the guides and other non-versioned content may be "polished up" for v3. Others may need more substantial rewrites. Some may be retired instead if they aren't widely used. Some may live on as v2 guides in perpetuity. We haven't really ironed out the details around exactly what that transition is going to look like, so I'm wary of making any promises with respect to that content.

Reply:
Yeah, that looks good and then no need to talk about guides and such. Then maybe just remove the "any" so it doesn't sound like 100% coverage? (Might be missed in a few corners somewhere..)